File: backend/src/main/java/com/back/fairytale/domain/keyword/dto/KeywordResponseDto.java
package com.back.fairytale.domain.keyword.dto;

import com.back.fairytale.domain.keyword.entity.Keyword;

public record KeywordResponseDto(
        Long keywordId,
        String keyword,
        String keywordType,
        int usageCount
) {
    public static KeywordResponseDto fromEntity(Keyword keyword) {
        return new KeywordResponseDto(
                keyword.getKeywordId(),
                keyword.getKeyword(),
                keyword.getKeywordType() != null ? keyword.getKeywordType().name() : null,
                keyword.getUsageCount()
        );
    }
}

File: backend/src/main/java/com/back/fairytale/domain/keyword/dto/KeywordResponseDto.kt
package com.back.fairytale.domain.keyword.dto

import com.back.fairytale.domain.keyword.entity.Keyword

data class KeywordResponseDto(
    val keywordId: Long,
    val keyword: String,
    val keywordType: String,
    val usageCount: Int
) {
    companion object {
        fun fromEntity(keyword: Keyword): KeywordResponseDto {
            return KeywordResponseDto(
                keywordId = keyword.keywordId ?: 0L,
                keyword = keyword.keyword,
                keywordType = keyword.keywordType?.name ?: "",
                usageCount = keyword.usageCount
            )
        }
    }
}
